src/DataLayouts/DataLayouts.jl

```diff
@@ -49,6 +49,13 @@ include("struct.jl")
 
 abstract type AbstractData{S} end
 
+abstract type AbstractDataSpecificCartesianIndex{N} <:
+              Base.AbstractCartesianIndex{N} end
+
+struct DataSpecificCartesianIndex{N} <: AbstractDataSpecificCartesianIndex{N}
```

Comment:
Can you please add a docstring here?

Reply:
These are internals, but I can add a doc string.